x/ibc/07-tendermint/types/update.go

```diff
@@ -63,14 +63,6 @@ func (cs ClientState) CheckHeaderAndUpdateState(
 
 // checkTrustedHeader checks that consensus state matches trusted fields of Header
 func checkTrustedHeader(header *Header, consState *ConsensusState) error {
-	if !header.TrustedHeight.EQ(consState.Height) {
```

Comment:
note: the consensus height was being retrieved using the the trusted height so this check was unnecessary